drivers/hud_apr/app/models/hud_apr/generators/shared/fy2024/base.rb

```diff
@@ -0,0 +1,826 @@
+            project_type: enrollment.project_type,
+            project_tracking_method: enrollment.project_tracking_method,
+            move_in_date: enrollment.move_in_date,
+          }
+        end
+        [pit_date, enrollments_for_date]
+      end.to_h.select { |_, v| v.present? }
+    end
+
+    delegate :client_scope, to: :@generator
+    delegate :end_date, to: :@report, prefix: :report
+
+    private def report_client_universe
+      HudApr::Fy2020::AprClient
+    end
+
+    private def report_living_situation_universe
+      HudApr::Fy2020::AprLivingSituation
+    end
+
+    private def report_ce_assessment_universe
+      HudApr::Fy2020::CeAssessment
+    end
+
+    private def report_ce_event_universe
+      HudApr::Fy2020::CeEvent
+    end
+
+    private def a_t
+      @a_t ||= report_client_universe.arel_table
+    end
+
+    private def ethnicities
+      {
+        '0' => {
+          order: 1,
+          label: 'Non-Hispanic/Non-Latin(a)(o)(x)',
+          clause: a_t[:ethnicity].eq(0),
+        },
+        '1' => {
+          order: 2,
+          label: 'Hispanic/Latin(a)(o)(x)',
+          clause: a_t[:ethnicity].eq(1),
+        },
+        '8 or 9' => {
+          order: 3,
+          label: 'Client Doesn\'t Know/Client Refused',
+          clause: a_t[:ethnicity].in([8, 9]),
+        },
+        '99' => {
+          order: 4,
+          label: 'Data Not Collected',
+          clause: a_t[:ethnicity].eq(99).or(a_t[:ethnicity].eq(nil)),
+        },
+        'Total' => {
+          order: 5,
+          label: 'Total',
+          clause: Arel.sql('1=1'),
+        },
+      }.sort_by { |_, m| m[:order] }.freeze
+    end
+
+    private def disability_clauses(suffix)
+      {
+        'Mental Health Disorder' => a_t["mental_health_problem_#{suffix}".to_sym].eq(1),
+        'Alcohol Use Disorder' => a_t["alcohol_abuse_#{suffix}".to_sym].eq(true).
+          and(a_t["drug_abuse_#{suffix}".to_sym].eq(false)),
+        'Drug Use Disorder' => a_t["drug_abuse_#{suffix}".to_sym].eq(true).
+          and(a_t["alcohol_abuse_#{suffix}".to_sym].eq(false)),
+        'Both Alcohol and Drug Use Disorders' => a_t["alcohol_abuse_#{suffix}".to_sym].eq(true).
+          and(a_t["drug_abuse_#{suffix}".to_sym].eq(true)),
+        'Chronic Health Condition' => a_t["chronic_disability_#{suffix}".to_sym].eq(1),
+        'HIV/AIDS' => a_t["hiv_aids_#{suffix}".to_sym].eq(1),
+        'Developmental Disability' => a_t["developmental_disability_#{suffix}".to_sym].eq(1),
+        'Physical Disability' => a_t["physical_disability_#{suffix}".to_sym].eq(1),
+      }
+    end
+
+    # NOTE: HMIS allows for clients to report multiple races. The APR however, does not, and has a single
+    # race field. The order that the races appear in the report is encoded in the 'order' of this hash.
+    # This practice is very brittle, so we'll copy those here and hard code those relationships
+    private def races
```

Comment:
Not sure if this is in use any more, but it will need to be updated with the new race categories.